include/onnxruntime/core/common/common.h

```diff
@@ -283,4 +283,10 @@ inline std::wstring ToWideString(const std::wstring& s) { return s; }
 inline std::string ToWideString(const std::string& s) { return s; }
 #endif
 
+#if ((__cplusplus >= 201703L) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L)))
```

Comment:
If we're moving to C++17 for good, why do we need this condition?

Reply:
Because CUDA 10.2 doesn't support C++17 and we still need to generate CUDA 10.2 packages, we must avoid to use C++17 in CUDA files(*.cu). Also, the *.h files that they include(like core/framework/float16.h).

Reply:
Ok. thanks. Would be good to document this somewhere.